File: src/migrations/migration7.ts
import {Kingdom} from '../kingdom/data/kingdom';
import {Migration} from './migration';
import {isNullable} from '../utils';

export class Migration7 extends Migration {
    constructor() {
        super(7);
    }

    override async migrateKingdom(game: Game, kingdom: Kingdom): Promise<void> {
        if (isNullable(kingdom.notes)) {
            kingdom.notes = {
                public: '',
                gm: '',
            };
        }
    }
}

File: src/jsMain/kotlin/at/posselt/pfrpg2e/migrations/migrations/Migration7.kt
package at.posselt.pfrpg2e.migrations.migrations

import at.posselt.pfrpg2e.kingdom.KingdomData
import at.posselt.pfrpg2e.kingdom.data.RawNotes
import com.foundryvtt.core.Game

class Migration7 : Migration(7) {
    override suspend fun migrateKingdom(game: Game, kingdom: KingdomData) {
        @Suppress("SENSELESS_COMPARISON")
        if (kingdom.notes == null) {
            kingdom.notes = RawNotes(
                public = "",
                gm = "",
            )
        }
    }
}